services/QuillLMS/app/services/classlink_integration/school_persister.rb

```diff
@@ -2,23 +2,36 @@
 
 module ClasslinkIntegration
   class SchoolPersister < ApplicationService
-    attr_reader :name, :district_id, :nces_id
+    attr_reader :classlink_instance_id, :district_id, :name, :nces_id, :sourced_id
 
-    def initialize(name:, district_id:, nces_id: nil)
+    def initialize(name:, district_id:, classlink_instance_id:, sourced_id:, nces_id: nil)
       @name = name
       @district_id = district_id
+      @classlink_instance_id = classlink_instance_id
       @nces_id = nces_id
+      @sourced_id = sourced_id
     end
 
     def run
+      ClasslinkInstanceSchool.find_or_create_by!(
+        classlink_instance_id:,
+        school_id: school.id
+      ) do |cis|
+        cis.external_id = sourced_id
+      end
+    rescue => e
+      msg = "Failed to persist school with sourcedId=#{sourced_id} and districtId=#{district_id}"
```

Comment:
I would add a custom error class for this as it will give better visibility with NR/sentry